src/app/store/courses/courses.selectors.ts

```diff
@@ -0,0 +1,14 @@
+import { createFeatureSelector, createSelector } from '@ngrx/store';
+
+import { coursesFeatureKey, CoursesState } from './courses.reducer';
+
+export namespace CoursesSelectors {
+  const selectCourses = createFeatureSelector<CoursesState>(coursesFeatureKey);
+
+  export const selectAllCourses = createSelector(selectCourses, (state: CoursesState) => state.courses);
+  export const selectCourse = createSelector(selectCourses, (state: CoursesState) => state.course);
```

Comment:
Why do we need this unused variable `selectCourse ` and `selectErrorMessage`?

Reply:
have removed